src/app/profiles/profiles.editor.modal.ts

```diff
@@ -33,148 +34,114 @@ import { TinyMCEPlugin } from '../rmm/plugin/tinymce.plugin';
 
 export class ProfilesEditorModalComponent {
     @Input() value: any[];
-    field_errors;
+    field_errors: Identity;
     allowed_domains = [];
     is_valid = false;
-    aliases_unique = [];
-    is_busy = false;
-    is_delete = false;
+
     is_update = false;
     is_create = false;
-    is_create_main = false;
-    type;
     is_visible_smtp_detail = false;
     is_different_reply_to = false;
     localpart;
     editor: any = null;
     selector: any;
     public tinymce_plugin: TinyMCEPlugin;
     constructor(
+        public profileService: ProfileService,
         public rmm: RMM,
         private location: Location,
         public snackBar: MatSnackBar,
         public dialog_ref: MatDialogRef<ProfilesEditorModalComponent>,
         public draftDeskservice: DraftDeskService,
-        @Inject(MAT_DIALOG_DATA) public data: any
+        @Inject(MAT_DIALOG_DATA) public identity: Identity
     ) {
         this.tinymce_plugin = new TinyMCEPlugin();
-        if (data && data.type) {
-            this.type = data.type;
-            delete data.type;
-        }
-        if (data.profile && data.profile.email) {
-            this.set_localpart(data);
-        }
-        if (!data || !Object.keys(data).length || !data.profile) {
-            data = { profile: {} };
+        if (!identity || !Object.keys(identity).length) {
+            identity = new Identity;
             const self = this;
```

Comment:
you don't need to create a `self` in this case, since the fat arrow `=>` retains `this` for you.

```javascript
identity.name = ['first_name', 'last_name'].map((attr) => {
  return this.profileService.me[attr];
}).join(' ');
```

Reply:
That'll be a left over hernan-ism .. *fixed*